### ISSUE TITLE
False positive for DL3010 for non tar archives

### ISSUE THREAD

**Issue by gautaz** [hadolint/hadolint#332]
Hello,

The current source code concerning DL3010 seems to consider that all files ending with:
https://github.com/hadolint/hadolint/blob/56fd3e1eed0e418b3a72dc619fc4282951d425e7/src/Hadolint/Rules.hs#L549
are tar archives.

The [Dockerfile ADD documentation](https://docs.docker.com/engine/reference/builder/#add) states:
> If `<src>` is a local tar archive in a recognized compression format...

I am not proficient in Haskell but a gzip/bzip2/xz archive is not necessarily a local tar archive.
Am I misunderstanding the behavior of this part of the source code?

Currently `hadolint` raises DL3010 when COPYing a bzip2 file even if this file is not a tar archive (and the Dockerfile ADD command will fail to extract such a file).

Referring to [the wikipedia tar page](https://en.wikipedia.org/wiki/Tar_(computing)#Suffixes_for_compressed_files), perhaps the right extensions to check would be something like:
```
archiveFormats = [".tar", ".tar.bz2", ".tb2", ".tbz", ".tbz2", ".tar.gz", ".tgz", ".tpz", ".tar.lz", ".tar.lzma", ".tlz", ".tar.xz", ".txz", ".tar.Z", ".tZ"]
```

It would be even better to check the real file format but I'm not sure if it is doable.

**Comment by SuperSandro2000** [comment]
We could use something like ``file``or check the magic number at the beginning.

**Comment by lorenzo** [comment]
Thanks, I'll update to use the suggested formats. We are not going to test for actual files since the would require actually executing the docker file.
